### ISSUE TITLE
problem with splitting `morphana` by `+`

### ISSUE THREAD

**Issue by sassbalint** [ril-lexknowrep/emCompound#3]
__Phenomenon:__

```
$ echo "+0,96%" | curl -F "file=@-" http://EMTSV_SERVER/tok/morph/pos | python3 emCompound/emCompound
...
  File "../emCompound/emCompound/emCompound.py", line 172, in process_sentence
    if morphemes[0]['mtag'] == PREVERB_POSTAG:
TypeError: In "no filename for stream" at 3: 'NoneType' object is not subscriptable
```

__Problem:__
[`morphana` can not be easily splitted by `+` marks](https://github.com/ril-lexknowrep/emCompound/blob/main/emCompound/emCompound.py#L163) if the token processed contains a `+` character itself.

__Investigation:__

Changing https://github.com/ril-lexknowrep/emCompound/blob/main/emCompound/emCompound.py#L172-L175 to
```python
    try:
        if morphemes[0]['mtag'] == PREVERB_POSTAG:
            preverb_flag = True
    except TypeError:
        msg = '\n'.join(f'{x}' for x in morphemes)
        raise RuntimeError(msg)
```

the result shows in detail what is happening:

```
None
<regex.Match object; span=(0, 5), match='[/N]='>
None
<regex.Match object; span=(0, 15), match='0[/Num|Digit]=0'>
<regex.Match object; span=(0, 15), match=',[/Num|Digit]=,'>
<regex.Match object; span=(0, 15), match='9[/Num|Digit]=9'>
<regex.Match object; span=(0, 15), match='6[/Num|Digit]=6'>
<regex.Match object; span=(0, 17), match='%[/N|Unit|Abbr]=%'>
<regex.Match object; span=(0, 6), match='[Nom]='>
```

which comes from the corresponding `morphana`:
```
+[/N]=++0[/Num|Digit]=0+,[/Num|Digit]=,+9[/Num|Digit]=9+6[/Num|Digit]=6+%[/N|Unit|Abbr]=%+[Nom]=
```

Splitting this by `+` marks we clearly get empty matches at 1st and 3rd splits.

The problem is caused by the not prudent design of the format of `morphana`.

__Solution?__

workaround 1: if the morpheme is `None` set its fields to empty string or something

workaround 2: delete all `+` chars from the corpus...

The general correct solution seems to be a hard task.

**Comment by gpetho** [comment]
I don't think proper compound tokens ever contain a '+' character. How about we add a condition to `emCompound` to ignore (= not try to split) tokens that contain a '+'? That seems like the correct solution (and the easiest to implement) to me.

**Comment by sassbalint** [comment]
Good idea, go ahead and thanks. :)

(Remark: this is also a workaround, not a general solution for splitting `morphana` correctly, but completely fine for us now.)

**Comment by gpetho** [comment]
> not a general solution for splitting `morphana` correctly

I agree, I think the general solution would be along the lines of inserting a line that splits `morphana` on the substring `+[/N]=+` and below that a foreach line that iterates over the return values between rows 162 and 163, and then `'+'.join()` the compound components established in each iteration of the `for m in morphemes` loop. This should yield the correct result for any real compounds that contain a '+', like 'Jazz+Az-koncert' or 'VIVA+-műsorvezető', assuming that emMorph can analyse these correctly. But these are so rare that I see little value in doing it this way.

**Comment by sassbalint** [comment]
An idea: splitting `readable` by `␣+␣` seems to be a good start for a general solution.

`readable` of `"+0,96%"` is correct:

```
+[/N] + 0[/Num|Digit] + ,[/Num|Digit] + 9[/Num|Digit] + 6[/Num|Digit] + %[/N|Unit|Abbr] + [Nom]
```
